### ISSUE TITLE
Save states partially break Jill of the Jungle

### ISSUE THREAD

**Issue by Matthaiks** [joncampbell123/dosbox-x#4141]
### Describe the bug

Using save states breaks the Help menu in Jill of the Jungle.

### Steps to reproduce the behaviour

1. Run "Jill of the Jungle: Episode 1" and start a new game.
2. Save a state and then load it.
3. Pressing F1 won't bring up the Help menu.

### Expected behavior

Loading save states should not affect the Help menu.

### What operating system(s) this bug have occurred on?

Windows 10 22H2 x64

### What version(s) of DOSBox-X have this bug?

DOSBox-X 2023.03.31 SDL1 MingGW

### Output log

```text
Saving state to slot: 1
File : .\save\CMOS is 43 bytes
File : .\save\CPU is 56852 bytes
File : .\save\DMA is 514 bytes
File : .\save\Dos is 6900 bytes
File : .\save\Dosbox-x is 14 bytes
File : .\save\EMS is 130 bytes
File : .\save\FPU is 35 bytes
File : .\save\IO handler is 0 bytes
File : .\save\Int10 is 83 bytes
File : .\save\IntTimer10 is 129 bytes
File : .\save\Joystick is 16 bytes
File : .\save\Keyboard is 107 bytes
File : .\save\Memory is 286932 bytes
File : .\save\Midi is 170 bytes
File : .\save\Mixer is 6148 bytes
File : .\save\Mouse is 119 bytes
File : .\save\Pic is 1129 bytes
File : .\save\Render is 466 bytes
File : .\save\Vga is 32042 bytes
File : .\save\XMS is 58 bytes
File : .\save\DOSBox-X_Version is 64 bytes
File : .\save\Program_Name is 5 bytes
File : .\save\Memory_Size is 4 bytes
File : .\save\Machine_Type is 7 bytes
File : .\save\Time_Stamp is 16 bytes
File : .\save\Save_Remark is 0 bytes
[15:42:26]: Saved. (Slot 1)
 Extracting: Program_Name
 Extracting: Time_Stamp
 Extracting: Save_Remark
Loading state from slot: 1
 Extracting: CMOS
 Extracting: DOSBox-X_Version
 Extracting: Program_Name
 Extracting: Memory_Size
 Extracting: Machine_Type
 Extracting: CPU
 Extracting: DMA
 Extracting: Dos
 Extracting: Dosbox-x
 Extracting: EMS
 Extracting: FPU
 Extracting: IO handler
 Extracting: Int10
 Extracting: IntTimer10
 Extracting: Joystick
 Extracting: Keyboard
 Extracting: Memory
 Extracting: Midi
 Extracting: Mixer
 Extracting: Mouse
 Extracting: Pic
 Extracting: Render
pixratio 1.200, dw true, dh false
Aspect ratio: 640 x 480  xToY=1.333 yToX=0.750
Screen report: Method 'Win98base' (1920.000 x 1080.000 pixels) at (0.000 x 0.000) (508.000 x 285.750 mm) (20.000 x 11.250 in) (96.000 x 96.000 DPI)
 Extracting: Vga
 Extracting: XMS
[15:42:28]: Loaded. (Slot 1)
```

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by joncampbell123** [comment]
Save states don't save much yet beyond the CPU, memory, and some video state, so this is no surprise.

Also games that use XMS/EMS like Wolfenstein 3D and Xargon will break with save states as well.

**Comment by Allofich** [comment]
Yes, I've also found save states are always a bit hit-and-miss. If you're playing a game and it has a built-in save and load feature, it is best to use that. Save states seem to usually work as long as the state you're trying to load to is not too different from the state you are trying to load from, like both are in gameplay mode (as opposed to, for example, loading from a menu into gameplay mode). In one game I played, if I remember correctly loading save states only worked if I was trying to load a state that had been saved in the same level I was loading from. In another game they usually worked but occasionally a "bad" one would be created that would freeze the game if loaded, so I alternated between multiple save slots.

**Comment by Matthaiks** [comment]
In case of Jill, DOSBox-X 0.84.0 2022-05-31 seems to have broken save states.

**Comment by MX9000** [comment]
You might try the other builds of the latest version of DOSBox-X.
With other games I have noticed that sometimes the save states work in some builds and in others they don't.
For example see bug #3754.

**Comment by Allofich** [comment]
If it was working before and stopped working, though, there may be something that needs fixing. If we can narrow it down to the commit that caused it to stop working, a fix will probably come soon after.

**Comment by Allofich** [comment]
I've been testing this, and I'm finding that the described issue happens with the `dosbox-x.exe` files packaged in the x64 SDL1 MingGW releases, but not those packaged in the vsbuild-win64 releases. This holds true both for the latest release and the May releases last year (so tentatively, I don't think DOSBox-X 0.84.0 2022-05-31 is related). @Matthaiks, did you find that the 2022-05-01 release worked but the 2022-05-31 release did not? Were you testing x64 SDL1 MingGW release? I found that even in the 2022-05-01 release, the x64 SDL1 MingGW version has the problem.

Meanwhile, the issue does not happen with the `dosbox-x.exe` from the "vsbuild-win64" configuration in the 2022-05-01, 2022-05-31, and 2023-03-31 releases.

Strangely, when testing with x64 SDL 1 `dosbox-x.exe` executables I created created with Visual Studio 2019 or 2022 via AppVeyor, the save state issue _does_ happen, whether using the latest code or code from the times of the 2022-05-01 or 2022-05-31 releases.

**Comment by Allofich** [comment]
The save state problem also occurs with a `dosbox-x.exe` I made locally with the latest code (Visual Studio 2022, SDL1, x64).

**Comment by Matthaiks** [comment]
@Allofich 2022-05-01 MinGW SDL1 x64 seems to work fine here (i.e. the Help menu shows up) unlike 2022-05-31 MinGW & VS SDL1 x64.

**Comment by Allofich** [comment]
I just re-tested, and in 2022-05-01 MinGW SDL1 x64 the Help menu does not show for me. I also tested with a default config (nothing in my dosbox-x.conf file but a mount line) and it still didn't work.

However... the 2022-05-01 MinGW SDL2 x64 version _does_ work. Just in case I had somehow taken a "good" save state with teh SDL2 version and a "bad" one with the other, I tried switching them, loading the SDL2 state with the SDL1 executable and vice-versa, but even then, the SDL1 .exe wouldn't show the menu with the SDL2 state, and the SDL2 .exe would show it with the SDL1 state.

@Matthaiks, are you sure you didn't use the SDL2 version? Have you tried using a default configuration?

**Comment by Matthaiks** [comment]
@Allofich
![DOSBox-X 0 83 25 (2023 04 06, 14-45-09)](https://user-images.githubusercontent.com/3577122/230382525-fc0bd0ca-9a9c-416d-ab58-52012ee770dc.png)
Default settings.

**Comment by Allofich** [comment]
In further testing, the 2022-05-01 MinGW SDL2 x64 version does not always work for me. I also tested the 2022-05-31 MinGW SDL2 x64 and 2023-03-31 MinGW SDL2 x64 releases, but they didn't work.

Quite messy results, so to recap:
My testing:
vsbuild-win64 releases seemingly always work
Other VS builds (VS 2019/2022 through AppVeyor, VS 2022 local) seemingly never work
MinGW SDL1 x64 releases seemingly never work
MinGW SDL2 x64 seems hit or miss with the 2022-05-01 release, and doesn't seem to work in the 2022-05-31 or 2023-03-31 releases.

Matthaiks testing:
MinGW SDL1 x64 2022-05-01 release works, MinGW SDL1 x64 2022-05-31 and 2023-03-31 releases don't

**Comment by maron2000** [comment]
I don't see any flaws on MinGW SDL1, SDL2 x64 (2023.03.31)
The capture below is MinGW SDL1.

https://user-images.githubusercontent.com/68574602/230400122-fd60d6a1-8929-4640-8d9e-618f3b25c9ae.mp4
